src/coreclr/tools/aot/ILCompiler.RyuJit/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -236,6 +237,53 @@ public void InitializeNonRelocationDependencies(DependencyList dependencies)
             _nonRelocationDependencies = dependencies;
         }
 
+        public IEnumerable<NativeSequencePoint> GetNativeSequencePoints()
+        {
+            var sequencePoints = new (string Document, int LineNumber)[_debugLocInfos.Length * 4 /* chosen empirically */];
+            try
+            {
+                foreach (var sequencePoint in _debugInfo.GetSequencePoints())
+                {
+                    int offset = sequencePoint.Offset;
+                    if (offset >= sequencePoints.Length)
+                    {
+                        int newLength = sequencePoints.Length;
+                        while (newLength <= offset)
```

Comment:
`Math.Max(2 * sequencePoints.Length, sequencePoint.Offset + 1)` instead of loop?